frontend/src/services/WorkspaceService.ts

```diff
@@ -114,24 +115,46 @@ export class WorkspaceService {
         return await client.getMyWorkspaceList();
     }
 
-    public static setCurrentWorkspace (workspace: Workspace | undefined) {
+    public static async setCurrentWorkspace (workspace: Workspace | undefined) {
         if ( workspace !== this._workspace ) {
+            LOG.debug("setCurrentWorkspace;", workspace)
             this._workspace = workspace;
             if ( this._observer.hasCallbacks(WorkspaceServiceEvent.CURRENT_WORKSPACE_CHANGED) ) {
+                if (workspace){
+
+                    // user, set current user !!!
+                    const workspaceId = workspace.id;
+                    const email: string | undefined = EmailAuthSessionService.getEmailAddress();
+
+                    LOG.debug("User email;", email)
+                    if (email && workspaceId) {
+                        const userList = await UserService.getWorkspaceUserList(workspaceId)
+
+                        if (userList  ) {
+                            const result = userList.filter(user => user.email === email)
+                            if (result ) {
+                               await UserService.setCurrentUser(result[0]);
+                           }
+                        } else {
+                            LOG.debug(`useSelectWorkspaceUserCallback: Did not find user to this email `, email);
+                        }
+                    }
+
+                }
                 this._observer.triggerEvent(WorkspaceServiceEvent.CURRENT_WORKSPACE_CHANGED);
             }
         }
     }
 
     private static async _initializeWorkspace () {
-        const list : readonly Workspace[] = await WorkspaceService.getMyWorkspaceList();
+      /*  const list : readonly Workspace[] = await WorkspaceService.getMyWorkspaceList();
```

Comment:
Don't leave commented code in the source code. It only makes these diffs look hard to read.